### PR TITLE
Fix handling of dedented continuation lines.

### PR DESCRIPTION
**Pull request by anntzer**
- Dedent the function definition before parsing it.
- Join continued-lines before attempting to parse the docstring.

Alternative to #441; further improves handling of parameter descriptions with continued lines (which were not handled correctly before).

Thanks for submitting a PR!

Please make sure to check for the following items:
- [X] Add unit tests and integration tests where applicable.  
      If you've added an error code or changed an error code behavior,
      you should probably add or change a test case file under `tests/test_cases/` and add 
      it to the list under `tests/test_definitions.py`.  
      If you've added or changed a command line option,
      you should probably add or change a test in `tests/test_integration.py`.
- [X] Add a line to the release notes (docs/release_notes.rst) under "Current Development Version".  
      Make sure to include the PR number after you open and get one.
   
Please don't get discouraged as it may take a while to get a review.

### CODE REVIEW

**Comment by samj1912** [comment]
Thanks for the patch. But from what I can see, this introduces new behaviour which is unrelated to the original issue. (It looks like if the doc string now has line continuation, it no longer errors with a D207). Although this might be a good improvement, can we please file a related issue to the repo with this behaviour and reference that instead?

Also it looks like most of the code in this PR is taken from https://github.com/PyCQA/pydocstyle/pull/441/

Can you either pick my commits from that PR to preserve the original code attribution or create a new PR with the improved handling of line continuations and update the docs to indicate this new behaviour?

**Comment by samj1912** [comment]
Looks like the issue this PR is solving is -  https://github.com/PyCQA/pydocstyle/issues/234

**Comment by anntzer** [comment]
The only reason I opened this separately is because you didn't pick up https://github.com/PyCQA/pydocstyle/pull/441#discussion_r414518686.  Now that you did I have no problem waiting for your PR to go in first and rebase this one over yours.

**Comment by samj1912** [comment]
The original PR was fixed. Would you like to rebase and update the description with the dedented bug fix?

**Comment by samj1912** [comment]
@anntzer bump?

**Comment by anntzer** [comment]
fixed :)

**Comment by anntzer** [comment]
What do you actually want to test for?

**Comment by samj1912** [comment]
That it doesn't throw D207 for normal doc string descriptions with line continuations

**Comment by anntzer** [comment]
done

File: docs/release_notes.rst [hunk 1]
```diff
@@ -20,6 +20,8 @@ Bug Fixes
   The bug caused some argument names to go unreported in D417 (#448).
 * Fixed an issue where skipping errors on module level docstring via #noqa
   failed when there where more prior comments (#446).
+* Support backslash-continued parameter descriptions in Numpy-style docstrings
```
Comment (samj1912) [inline docs/release_notes.rst:23]: Isn't it doing this for more than just numpy parameter descriptions at this point?

Comment (samj1912) [reply docs/release_notes.rst:23]: should we update this?

**Comment by anntzer** [comment]
thanks for fixing the last part :)

**Pull request merged**